src/cdk-experimental/selection/selection.spec.ts

```diff
@@ -0,0 +1,691 @@
+         expect(component.cdkSelection.isSelected('durian', 3)).toBeFalsy();
+         expect(component.getSelectionToggle(1).textContent.trim()).toBe('unchecked');
+         expect(component.getSelectionToggle(2).textContent.trim()).toBe('unchecked');
+         expect(component.getSelectionToggle(3).textContent.trim()).toBe('unchecked');
+       }));
+
+    it('should de-selected selected item when clicked', fakeAsync(() => {
+         // Make all items selected.
+         component.clickSelectAll();
+
+         component.clickSelectionToggle(1);
+
+         expect(component.cdkSelection.isSelected('banana', 1)).toBeFalsy();
+         expect(component.getSelectionToggle(1).textContent.trim()).toBe('unchecked');
+
+         // And all the others are not affected.
+         expect(component.cdkSelection.isSelected('apple', 0)).toBeTruthy();
+         expect(component.cdkSelection.isSelected('cherry', 2)).toBeTruthy();
+         expect(component.cdkSelection.isSelected('durian', 3)).toBeTruthy();
+         expect(component.getSelectionToggle(0).textContent.trim()).toBe('checked');
+         expect(component.getSelectionToggle(2).textContent.trim()).toBe('checked');
+         expect(component.getSelectionToggle(3).textContent.trim()).toBe('checked');
+       }));
+
+    it('should emit the correct selection change events', fakeAsync(() => {
+         component.clickSelectionToggle(1);
+
+         expect(component.selectionChange!.before).toEqual([]);
+         expect(component.selectionChange!.after).toEqual([
+           {value: 'banana', index: 1},
+         ]);
+
+         component.clickSelectionToggle(2);
+
+         expect(component.selectionChange!.before).toEqual([
+           {value: 'banana', index: 1},
+         ]);
+         expect(component.selectionChange!.after).toEqual([
+           {value: 'banana', index: 1},
+           {value: 'cherry', index: 2},
+         ]);
+
+         component.clickSelectionToggle(2);
+
+         expect(component.selectionChange!.before).toEqual([
+           {value: 'banana', index: 1},
+           {value: 'cherry', index: 2},
+         ]);
+         expect(component.selectionChange!.after).toEqual([
+           {value: 'banana', index: 1},
+         ]);
+       }));
+  });
+});
+
+describe('CdkSelection with multiple = false', () => {
+  let fixture: ComponentFixture<SingleSelectionApp>;
+  let component: SingleSelectionApp;
+
+  beforeEach(async(() => {
+    TestBed
+        .configureTestingModule({
+          imports: [CdkSelectionModule],
+          declarations: [
+            SingleSelectionApp,
+          ]
+        })
+        .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SingleSelectionApp);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should uncheck the previous selection when selecting new item', fakeAsync(() => {
+       // Everything start as unchecked.
```

Comment:
nit: indent here should be +2 rather than +4

Reply:
That is odd, I had to fight `yarn format` to make it +2 instead of +4 for those `fakeAsync` blocks.

Reply:
Ah yeah, clang has issues with blocks like that (it's why we don't enforce the auto-formatter)